tests/run/tryPatternMatch.check

```diff
@@ -0,0 +1,9 @@
+IllegalArgumentException: abc
+IllegalArgumentException
+IOException
```

Comment:
this shows a bug) `IOException` should NOT be printed in this testcase(guard is false))